src/topp/DecoyDatabase.cpp

```diff
@@ -96,6 +108,15 @@ class TOPPDecoyDatabase :
     setValidStrings_("enzyme", all_enzymes);
 
     registerSubsection_("Decoy", "Decoy parameters section");
+
+    // New options for neighbor peptide search
+    registerTOPPSubsection_("Neighbor_Search", "Parameters for neighbor peptide search");
+    registerInputFile_("Neighbor_Search:neighbor_in", "<file>","", "Input FASTA file for neighbor peptide search", false);
+    setValidFormats_("Neighbor_Search:neighbor_in", {"fasta"});
+    registerStringOption_("Neighbor_Search:resolution", "<choice>", "low", "Sets the resolution of the neighbor peptide search. Options are 'low'(0.05 Da) and 'high'(1.0005079 Da)",false);
+    setValidStrings_("Neighbor_Search:resolution", ListUtils::create<String>("low,high"));
+    registerDoubleOption_("Neighbor_Search:mass_tolerance", "<double>", 0.00004, "Tolerance of the mass of the neighbor peptide m1-m2 > tm", false, true);
+    registerDoubleOption_("Neighbor_Search:ion_tolerance", "<double>", 0.25, "Tolerance of the proportion of b and y ions shared by the neighbor peptide 2*B12/B1+B2 > ti", false, true);
```

Comment:
use a name which is clearer, see the discussion for the parameter of the function in your class